altapay.php

```diff
@@ -1,544 +1,664 @@
-            } else if ($captured == $refunded && $refunded == $reserved || $refunded == $reserved) {
-                $order->update_status('refunded');
-                $releaseResult = $api->releaseReservation($txnID);
-                if (!$releaseResult->wasSuccessful()) {
-                    $order->add_order_note(__('Release failed: ' . $releaseResult->getMerchantErrorMessage(), 'altapay'));
-                    echo json_encode(array('status' => 'error', 'message' => $releaseResult->getMerchantErrorMessage()));
-                }
-            } else {
-                $order->update_status('processing');
-                $releaseResult = $api->releaseReservation($txnID);
-                if (!$releaseResult->wasSuccessful()) {
-                    $order->add_order_note(__('Release failed: Order cannot be released', 'altapay'));
-                    echo json_encode(array('status' => 'error', 'message' => $releaseResult->getMerchantErrorMessage()));
-                }
-            }
-        } catch (Exception $e) {
-            return WP_Error('error', "Could not login to the Merchant API: " . $e->getMessage());
-        }
-        wp_die();
-    }
+// Load AltaPay API Library
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'classes' . DIRECTORY_SEPARATOR . 'core' . DIRECTORY_SEPARATOR . 'AltapaySettings.php';
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'helpers' . DIRECTORY_SEPARATOR . 'AltapayHelpers.php';
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'includes' . DIRECTORY_SEPARATOR . 'CreditCardTokenControl.php';
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'includes' . DIRECTORY_SEPARATOR . 'OrderStatusChangeActions.php';
+
+/**
+ * Init AltaPay settings and gateway
+ */
+function init_altapay_settings() {
+	// Make sure WooCommerce and WooCommerce gateway is enabled and loaded
+	if ( ! class_exists( 'WC_Payment_Gateway' ) ) {
+		return;
+	}
+	$settings = new AltapaySettings();
+	// Add Gateway to WooCommerce if enabled
+	if ( json_decode( get_option( 'altapay_terminals_enabled' ) ) ) {
+		add_filter( 'woocommerce_payment_gateways', 'altapay_add_gateway' );
+	}
+
+	/**
+	 * Add gateways to WooCommerce.
+	 *
+	 * @param $methods
+	 *
+	 * @return array
+	 */
+	function altapay_add_gateway( $methods ) {
+		$pluginDir = plugin_dir_path( __FILE__ );
+		// Directory for the terminals
+		$terminalDir = $pluginDir . 'terminals/';
+		// Temp dir in case the one from above is not writable
+		$tmpDir = sys_get_temp_dir();
+		// Get enabled terminals
+		$terminals = json_decode( get_option( 'altapay_terminals_enabled' ) );
+
+		if ( $terminals ) {
+			foreach ( $terminals as $terminal ) {
+				// Load Terminal information
+				$terminalInfo = json_decode( get_option( 'altapay_terminals' ) );
+				$terminalName = $terminal;
+				foreach ( $terminalInfo as $term ) {
+					if ( $term->key === $terminal ) {
+						$terminalName   = $term->name;
+						$terminalNature = $term->nature;
+						if ( in_array( 'CreditCard', $terminalNature ) ) {
```

Comment:
Always strict compare strings
```suggestion
						if ( in_array( 'CreditCard', $terminalNature, true ) ) {
```